lib/mobile_app_backend_web/controllers/schedule_controller.ex

```diff
@@ -39,20 +56,50 @@ defmodule MobileAppBackendWeb.ScheduleController do
     json(conn, response)
   end
 
-  @spec get_filter(String.t(), String.t()) :: [JsonApi.Params.filter_param()]
-  defp get_filter(stop_ids, date_time) do
-    date_time = Util.parse_datetime!(date_time)
-    service_date = Util.datetime_to_gtfs(date_time)
-    [stop: stop_ids, date: service_date]
+  @spec parse_service_date(String.t()) :: Date.t()
+  defp parse_service_date(date_string) do
+    date_string
+    |> Util.parse_datetime!()
+    |> Util.datetime_to_gtfs()
+  end
+
+  @spec get_filter(String.t(), Date.t()) :: [JsonApi.Params.filter_param()]
+  defp get_filter(stop_id, service_date) do
+    [stop: stop_id, date: service_date]
+  end
+
+  @spec fetch_schedules_parallel([[JsonApi.Params.filter_param()]]) ::
+          %{schedules: [MBTAV3API.Schedule.t()], trips: JsonApi.Object.trip_map()} | :error
+  defp fetch_schedules_parallel(filters) do
+    filters
+    |> Task.async_stream(fn filter_params ->
+      {filter_params, fetch_schedules(filter_params)}
+    end)
+    |> Enum.reduce_while(%{schedules: [], trips: %{}}, fn result, acc ->
+      case result do
+        {:ok, {_params, %{schedules: schedules, trips: trips}}} ->
+          {:cont, %{schedules: acc.schedules ++ schedules, trips: Map.merge(acc.trips, trips)}}
```

Comment:
This concatenation means that the response will no longer properly be sorted by departure time. If that sort order is required downstream, we should preserve it, and if it is not, it might make sense to remove it from `fetch_schedules/1`.

Reply:
The list of schedules is ultimately converted to a map [here](https://github.com/mbta/mobile_app/blob/0832cbc72be3c5a3bf7389ae7c564efe0255127c/shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/UpcomingTrip.kt#L104), so it seems safe to remove the sorting. Will run the app locally against dev-orange to confirm.